Checking initial version of README for Python Binding APIs.

KrishnaKolakaluri opened the pull request.

KrishnaKolakaluri committed:
Update README.md

KrishnaKolakaluri committed:
Syncing with upstream.Merge remote-tracking branch 'upstream/master'

duobliu1 commented on README.md:
````diff
@@ -0,0 +1,258 @@
+                 ("serial",ctypes.c_char * 64)]                 
+```
+```
+def set_fan_rpm(fanid, rpm):
````
Define fan as class with id as one of the data members.  set_rpm(rpm) as setter only takes rpm as input param.

duobliu1 commented on README.md:
```diff
@@ -0,0 +1,258 @@
+
+
+       --Returns number of leds that are available in the system. The ids start from 1 to num_of_leds.  
+led = get_led_info(id)
```
rename to get_led()

duobliu1 commented on README.md:
````diff
@@ -0,0 +1,258 @@
+         Char:
+       }
+      ``` 
+led.set_led(state)
````
rename to set_state()

duobliu1 commented on README.md:
```diff
@@ -0,0 +1,258 @@
+
+       --Sets the led state to either on or off.  
+       
+led.set_led_color(color)
```
rename to set_color()

duobliu1 commented on README.md:
```diff
@@ -0,0 +1,258 @@
+
+       --Sets the led color.
+              
+led.set_led_char(char)
```
rename to set_char()

duobliu1 commented on README.md:
```diff
@@ -0,0 +1,258 @@
+
+
+      --Returns number of fans that are available in the system. The ids start from 1 to num_of_fans.       
+fan  = get_fan_info(id)  
```
rename to get_fan()

duobliu1 commented on README.md:
````diff
@@ -0,0 +1,258 @@
+       SN:     NULL
+      }
+      ```
+fan.set_fan_rpm(rpm)  
````
rename to set_rpm()

duobliu1 commented on README.md:
```diff
@@ -0,0 +1,258 @@
+
+
+       --Returns number of sensors that are available in the system. The ids start from 1 to num_of_therm_sensors.
+thermal = get_thermal_info(id)
```
get_thermal()

duobliu1 commented on README.md:
```diff
@@ -0,0 +1,258 @@
+     --Sets the fan's speed in RPMs.       
+     
+     
+fan.set_fan_percentage(percent)  
```
set_percentage

duobliu1 commented on README.md:
```diff
@@ -0,0 +1,258 @@
+
+    --Sets the fan's speed as a percentage.  
+    
+fan.set_fan_mode(mode)  
```
set_mode

duobliu1 commented on README.md:
```diff
@@ -0,0 +1,258 @@
+    --Sets the fan's speed by mode.  
+    
+    
+fan.set_fan_direction(dir)
```
set_direction

duobliu1 commented on README.md:
```diff
@@ -0,0 +1,258 @@
+
+      --Return number of ports that are {Q}SFP-capable.
+
+sfp = get_sfp_info(port)
```
get_sfp()

duobliu1 commented on README.md:
```diff
@@ -0,0 +1,258 @@
+
+
+       --Returns number of PSUs that are available in the system. The ids start from 1 to num_of_psus.
+psu = get_psu_info(id)
```
get_psu()

duobliu1 commented on README.md:
````diff
@@ -0,0 +1,258 @@
+  num_fans = num_fans - 1
+```
+
+In the code example above, libonlp is the python library which internally defines the functions ‘get_number_fans'.  This function provide total number of fans available in the system. 
````
"define the functions 'get_number_fans'" ---> "define the function of get_number_of_fans()"

duobliu1 commented on README.md:
```diff
@@ -0,0 +1,258 @@
+
+In the code example above, libonlp is the python library which internally defines the functions ‘get_number_fans'.  This function provide total number of fans available in the system. 
+
+We get the number of fans with 'fan_info_get(num_fans)' and loop through the count while setting fan RPM, and printing out the fan description.
```
"loop through the count while setting fan RPM, and printing...." --> "loop through all the fans to set RPM, and print  ....."

KrishnaKolakaluri committed:
Update README.md

KrishnaKolakaluri committed:
Update README.md

KrishnaKolakaluri committed:
Update README.md

Defined an example 'class fan' and updated the API reference accordingly for all other objects.

Commit:
Update README.md

duobliu1 commented on README.md:
```diff
@@ -0,0 +1,237 @@
+    onlp_fan = onlp_fan_info_t()
+    def __init__(self, fanid):
+        self.fanoid = self.fanoid | fanid
+        self.libfan = ctypes.CDLL('/lib/x86_64-linux-gnu/libonlp.so')
```
we can put dll loading into SDK init() so no need to load it each time create object.

Commit:
Moved t

Commit:
Update README.md

Commit:
Update README.md

Commit:
Update README.md
